### PR TITLE
[Fleet] Expose agent privileges root in package endpoint

### PR DESCRIPTION
**Pull request by nchaulet**
To be able to display root requirement in the integration details page in Fleet we need to add package and datastream agent privileges in the package endpoint.

That PR does that.

## Todo

- [x] Add tests

### CODE REVIEW

File: packages/testdata/marshaler/packages.json [hunk 1]
```diff
@@ -34,6 +34,53 @@
...
+   "title": "Agent Privileges",
+   "version": "1.0.0",
+   "release": "beta",
+   "description": "Test package-specified agent privileges",
+   "type": "solution",
+   "download": "/epr/agent_privileges/agent_privileges-1.0.0.zip",
+   "path": "/package/agent_privileges/1.0.0",
+   "conditions": {
+    "kibana": {
+     "version": "\u003e=7.16.0"
```
Comment (jsoriano) [inline packages/testdata/marshaler/packages.json:48]: Umm, this looks weird. But not new in any case.

**Review by jsoriano — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,68 +13,66 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 ## [v1.22.0](https://github.com/elastic/package-registry/compare/v1.21.0...v1.22.0)
 
 ### Breaking changes
 
 ### Bugfixes
 
-* Update Go runtime to 1.21.3. [#1102](https://github.com/elastic/package-registry/pull/1102)
-* Raise an error if the value of environment variables used to set parameters are not valid [#1103](https://github.com/elastic/package-registry/pull/1103)
+- Update Go runtime to 1.21.3. [#1102](https://github.com/elastic/package-registry/pull/1102)
+- Raise an error if the value of environment variables used to set parameters are not valid [#1103](https://github.com/elastic/package-registry/pull/1103)
```
Comment (jsoriano) [inline CHANGELOG.md:29]: Umm, why these changes? Do we need to do them?

Comment (nchaulet) [reply CHANGELOG.md:29]: probably not my IDE did some formating on save will revert that

File: packages/testdata/marshaler/packages.json (hunk 1 above)
Comment (jsoriano) [reply packages/testdata/marshaler/packages.json:48]: This seems an issue with these tests, registry API responses don't do this encoding.
